src/blockchain/Blockchain.ts

```diff
@@ -82,48 +96,115 @@ export default class Blockchain implements IBlockchainApi {
     identity: Identity,
     tx: SubmittableExtrinsic
   ): Promise<TxStatus> {
-    const accountAddress = identity.address
-    const nonce = await this.getNonce(accountAddress)
-    const signed: SubmittableExtrinsic = identity.signSubmittableExtrinsic(
-      tx,
-      nonce.toHex()
-    )
-    log.info(`Submitting ${tx.method}`)
-
-    return new Promise<TxStatus>((resolve, reject) => {
-      signed
-        .send(result => {
-          log.info(`Got tx status '${result.status.type}'`)
-
-          const { status } = result
-          if (ErrorHandler.extrinsicFailed(result)) {
-            log.warn(`Extrinsic execution failed`)
-            log.debug(`Transaction detail: ${JSON.stringify(result, null, 2)}`)
-            const extrinsicError: ExtrinsicError =
-              this.errorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
-
-            log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
-            reject(extrinsicError)
-          }
-          if (status.type === 'Finalized') {
-            resolve(new TxStatus(status.type))
-          } else if (status.type === 'Invalid' || status.type === 'Dropped') {
-            reject(new Error(`Transaction failed with status '${status.type}'`))
-          }
-        })
-        .catch((err: Error) => {
-          // just reject with the original tx error from the chain
-          reject(err)
+    try {
+      const nonce: Index = await this.getNonce(identity.address)
+      if (nonce.toNumber() === -1) {
+        throw new Error('Error while retrieving Nonce')
+      } else {
+        const signed = identity.signSubmittableExtrinsic(tx, nonce.toHex())
+        log.info(`Submitting ${tx.method} with Nonce ${nonce}`)
+
+        return new Promise<TxStatus>((resolve, reject) => {
+          signed
+            .send(result => {
+              log.info(`Got tx status '${result.status.type}'`)
+
+              const { status } = result
+              if (ErrorHandler.extrinsicFailed(result)) {
+                log.warn(`Extrinsic execution failed`)
+                log.debug(
+                  `Transaction detail: ${JSON.stringify(result, null, 2)}`
+                )
+                const extrinsicError: ExtrinsicError =
+                  this.errorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
+
+                log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
+                reject(extrinsicError)
+              }
+              if (status.type === 'Finalized') {
```

Comment:
I have added constants for status.

Reply:
yes thanks for the heads up, resolved the conflicts